### DISCUSSION TITLE
Generically use ModelSelect types

### DISCUSSION THREAD

**Discussion by dariusj18** [prisma/prisma#9591] [Q&A]
Say I have a `User` model and that in turn generates a `Prisma.UserSelect` type.  I would like to use the same select in multiple places and have therefore created an object like 

```typescript
const defaultSelect = {
  id: true,
  name: true,
}
```

and I want to use it like this

```typescript
const users = prisma.users.findMany({
  select: {
    ...defaultSelect
  }
});
```

this works, however I would like to type the defaultSelect like this 
```typescript
const defaultSelect: Prisma.UserSelect = {
  id: true,
  name: true,
}
```

but I get an error `Type '{}' is missing the following properties from type 'UserResponseDto': id, name.` because typescript doesn't seem to recognize that the typed object is the same as the implied typed object.

Is there any trick I can use to make it easier to get code hints while editing the defaultSelect while also not getting the type error?

**Comment by ryands17** [comment]
@dariusj18 👋 

Could you share what is `UserResponseDto` in this case? The type above works fine for me:

![image](https://user-images.githubusercontent.com/19697099/135973559-6e6c7997-7930-4c32-ac9b-8841d93035c6.png)

**Reply by dariusj18** [reply]
Here's a more thorough example.

```prisma
model Test {
  id   String    @id @default(uuid()) @db.Uuid
  name String
}
}
```

```typescript
import { TestSub } from '@prisma/client';

export class TestDto {
  id: string;
  name: string;
}

const testSelect: Prisma.TestSelect = {
  id: true,
  name: true,
};

async test(): Promise<TestDto> {
  const tests = this.prisma.test.findFirst({
    select: {
      ...testSelect,
    }
  });
  return tests;
}
```

![image](https://user-images.githubusercontent.com/10720/136037407-140bc846-2d56-4524-acc4-da9f3c82bdac.png)

But if I remove the `Prisma.TestSelect` type from `testSelect` I get no error.

**Reply by ryands17** [reply]
Prisma doesn't return classes so you can't assign a `class` as the return type. You would need to do something like this (classes won't work):

```ts
const testSelect: Prisma.TestSelect = {
  id: true,
  name: true,
};

type TestDto = Prisma.TestGetPayload<{
  select: {
    id: true,
    name: true
  }
}>

async test(): Promise<TestDto> {
  const tests = this.prisma.test.findFirst({
    select: {
      ...testSelect,
    }
  });
  return tests;
}
```

**Reply by dariusj18** [reply]
It doesn't return a class, however it does return an object that matches the class definition, which as far as I can tell works fine for typescript (and is an important thing from a development PoV).

**Reply by ryands17** [reply]
I don't think both are the same. Instances of that class would expect some sort of methods or something present on that and TypeScript does complain. So I don't that's possible to assign a class directly to the return type. You would need to have an intermediate mapper that converts the data returned from Prisma to a class like representation.

**Reply by dariusj18** [reply]
I believe you are misinterpreting the issue. If I remove the type from testSelect everything works as expected. My Dto has no methods and using a dumb Object for this situation is a common pattern.

**Reply by ryands17** [reply]
What are you passing in `select` then instead of `testSelect`?

**Reply by dariusj18** [reply]
I am still passing in `...testSelect`, it just isn't typed.  ex, this is fine:

```
import { TestSub } from '@prisma/client';

export class TestDto {
  id: string;
  name: string;
}

const testSelect = {
  id: true,
  name: true,
};

async test(): Promise<TestDto> {
  const tests = this.prisma.test.findFirst({
    select: {
      ...testSelect,
    }
  });
  return tests;
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
